### PR TITLE
Features/request api for customers

### PR DESCRIPTION
**Pull request by alan2112000**
Finish SyncCustomersService

 1. SyncCustomersService is responsible for sync user from the response
    find the original user to sync data or create user 

2. WebApiService is responsible for the HTTP get method
 and only request and return the response

3.  

 - Routine rake is for cron job to routiinely execute the rake
    the fetch time will set from the setting.yml

 -  In the deploy first time to sync data it's better to set cron job
    execute every 15mintutes and set the time to 30mintues to avoid the
    internet problem

### CODE REVIEW

**Comment by alan2112000** [comment]
## self reminder

> rebase and modify the commit message and log before `merge`

File: config/application.rb [hunk 1]
```diff
@@ -26,5 +26,6 @@ class Application < Rails::Application
     # Dir[File.join(Rails.root, 'lib/**/*.rb')].each { |file| require file }
 
     config.autoload_paths << Rails.root.join('lib')
+    ::APP_CONFIG = ActiveSupport::HashWithIndifferentAccess.new(config_for(:setting))
```
Comment (GGD) [inline config/application.rb]: might be better putting setting things to `config/initializers`, e.g. `config/initializers/app_config.rb`

Comment (alan2112000) [inline config/application.rb]: yup it's better !

**Comment by GGD** [comment]
:ok_hand:

**Pull request merged**